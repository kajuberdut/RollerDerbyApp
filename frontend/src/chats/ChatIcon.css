
.ChatIcon {
    /* font-size: 30px; */
    /* margin: 5px; */
    padding: 10px;
    width: 55px;
    height: 55px;
    /* border: 2px solid black; */
    border: 1px solid #ccc;
    border-radius: 50%;
    background: radial-gradient(circle, #f5f5f5, #e5e5e5);
    box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.1); 
}

.ChatIcon-Icon {
  font-size: 30px;
  padding-top: 4px;
  padding-left: 1px;
  color: black;
}

.ChatIcon:hover {
  /* text-decoration: none;
  color: rgb(29, 27, 29); */
  /* font-size: 40px; */
  width: 70px;
  height: 70px;

}

.ChatIcon:hover .ChatIcon-Icon {
  font-size: 40px;
  padding-top: 6px;
  padding-left: 1px;
}

