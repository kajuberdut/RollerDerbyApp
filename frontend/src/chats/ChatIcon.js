import React, { Component } from 'react';
// import { FaHeart, FaRegHeart } from 'react-icons/fa';
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faCommentAlt } from "@fortawesome/free-solid-svg-icons";
// import { star} from "@fortawesome/free-solid-svg-icons"
import "./ChatIcon.css";

function ChatIcon() {
    return (
    <div className='ChatIcon' role="ChatIcon">
      {/* <FontAwesomeIcon icon={faCommentAlt} style={{fontSize: '25px', marginRight: '7px'}}/> */}
      <FontAwesomeIcon icon={faCommentAlt} className='ChatIcon-Icon'/>
      {/* <FontAwesomeIcon icon={Star} style={{fontSize: '50px'}}/> */}

    </div>
    ) 
  }


export default ChatIcon