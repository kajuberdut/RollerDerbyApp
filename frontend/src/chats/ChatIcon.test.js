import { render, screen } from '@testing-library/react';
import ChatIcon from './ChatIcon';

// npm test ChatIcon.test.js
// must be in frontend directory 

test('renders chat icon page', () => {
  render(<ChatIcon />);
  const icon = screen.getByRole('ChatIcon');
  expect(icon).toBeInTheDocument();
});


test('matches snapshot', function() {
  const { asFragment } = render(<ChatIcon />);
  expect(asFragment()).toMatchSnapshot(); 
})

