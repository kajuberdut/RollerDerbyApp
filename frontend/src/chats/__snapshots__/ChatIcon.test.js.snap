// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ChatIcon"
    role="ChatIcon"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-message ChatIcon-Icon"
      data-icon="message"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M64 0C28.7 0 0 28.7 0 64V352c0 35.3 28.7 64 64 64h96v80c0 6.1 3.4 11.6 8.8 14.3s11.9 2.1 16.8-1.5L309.3 416H448c35.3 0 64-28.7 64-64V64c0-35.3-28.7-64-64-64H64z"
        fill="currentColor"
      />
    </svg>
  </div>
</DocumentFragment>
`;
